pymodbus/server/simulator/http_server.py

```diff
@@ -374,12 +373,12 @@ def helper_build_html_calls_submit_set(self, params):
     def helper_build_html_calls_submit(self, params):
         """Build html calls submit."""
         call_foot = ""
-        if params["submit"] == "Clear":
-            self.call_list = []
-            call_foot = "Cleared list, monitoring active"
-        elif params["submit"] == "Stop":
+        if params["submit"] == "Stop":
             self.call_monitor = CallTypeMonitor()
             call_foot = "Stopped monitoring"
+        # elif params["submit"] == "Clear":
```

Comment:
I preserved this logic since there seems to be an intended use of `call_list` but I don't know what it is yet.  It should either be finished or removed entirely.

Reply:
Why do you remove the submit function "Clear" that is a response to a http request, that is needed. It clears a listbox in the html.

Reply:
`call_list` is never used anywhere in the code, besides being set to `[]`.  I think some logic might be missing.

Reply:
Yes simulator is work in progress as noted in TODO.md, but that does not mean it is correct to remove it.

Reply:
In that case, when you fully implement `call_list` you'll have to type it as `List[<Something>]`.  Neither mypy nor I can guess the future.

Reply:
You are right, and that can be done now as list[str], but removing the code is not a particular friendly approach.

Reply:
Resolved in rebase